lib/common.py

```diff
@@ -329,9 +351,10 @@ def yum_remove_added_packages(self, base_packages):
             print("No added packages to remove.")
 
     def yum_restore_saved_state(self):
-        """ Currently only able to remove added packages """
-        assert self.saved_packages_list is not None, "Can't restore previous state without a package list"
-        self.yum_remove_added_packages(self.saved_packages_list)
+        """ Restore yum state to saved state """
+        assert self.saved_packages_list is not None and self.saved_rollback_id, "Can't restore previous state without a package list"
+        self.ssh(['yum', 'history', 'rollback', '--enablerepo=*', self.saved_rollback_id, '-y'])
+        assert self.saved_packages_list == self.packages()
```

Comment:
If it fails, the output will be very hard to read. I would prefer a diff and assert that the diff is empty.

Reply:
What about keeping this assert, with an error message ?

Reply:
Diffing would add algo complexity to the tests

Reply:
I think it's worth it to make the output usable. By the way you don't need to do the detailed comparison before you detect that there's a difference. When it's OK, the complexity will not be higher.